universal-application-tool-0.0.1/app/services/export/JsonExporter.java

```diff
@@ -0,0 +1,148 @@
+package services.export;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.jayway.jsonpath.DocumentContext;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import models.Application;
+import repository.ProgramRepository;
+import services.CfJsonDocumentContext;
+import services.PaginationResult;
+import services.PaginationSpec;
+import services.Path;
+import services.applicant.AnswerData;
+import services.applicant.ApplicantService;
+import services.applicant.JsonPathProvider;
+import services.applicant.ReadOnlyApplicantProgramService;
+import services.applicant.question.CurrencyQuestion;
+import services.applicant.question.MultiSelectQuestion;
+import services.applicant.question.NumberQuestion;
+import services.program.ProgramDefinition;
+import services.question.LocalizedQuestionOption;
+
+/** Exports all applications for a given program as JSON. */
+public class JsonExporter {
+
+  private final ApplicantService applicantService;
+  private final ProgramRepository programRepository;
+
+  @Inject
+  JsonExporter(ApplicantService applicantService, ProgramRepository programRepository) {
+    this.applicantService = checkNotNull(applicantService);
+    this.programRepository = checkNotNull(programRepository);
+  }
+
+  public String export(ProgramDefinition programDefinition) {
+    DocumentContext jsonApplications = makeEmptyJsonArray();
+    PaginationResult<Application> applicationPaginationResult =
+        programRepository.getApplicationsForAllProgramVersions(
+            programDefinition.id(), PaginationSpec.MAX_PAGE_SIZE_SPEC, Optional.empty());
```

Comment:
curious about pagination here?

Reply:
this sets the page size to max integer (https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/services/PaginationSpec.java#L6).